src/SubscriberBase.h

```diff
@@ -18,34 +30,115 @@ class SubscriberBaseT : public Subscriber<T>,
   virtual void onCompleteImpl() = 0;
   virtual void onErrorImpl(folly::exception_wrapper ex) = 0;
 
+  // used to be able to cancel subscription immediatelly, making sure we dont
+  // deliver any other signals after that
+  // also to break the reference cycle involving storing subscription pointer
+  // for the users of the SubscriberBase
+  class SubscriptionShim : public Subscription {
```

Comment:
can you elaborate more on the proposal? What would be the inheritance hierarchy?
currently the structure is this:
ClientsMySubscriber : derives from SubscriberBase

I didn't want to have something like
CancellationDecorator : derives from (ClientMySubscriber : derives from SubscriberBase)

I wanted to have the client code class to be the terminal leaf in the inheritance.